libs/langchain/langchain/chains/base.py

```diff
@@ -63,13 +63,10 @@ def invoke(
         **kwargs: Any,
     ) -> Dict[str, Any]:
         config = config or {}
-        return self(
-            input,
-            callbacks=config.get("callbacks"),
-            tags=config.get("tags"),
-            metadata=config.get("metadata"),
-            **kwargs,
-        )
+        config_kwargs: Dict = {
```

Comment:
LHS is simpler and more readable in this case, I would suggest to either revert or else create a helper function that does a projection operation.

```
project_dict(config, ['callbacks', 'tags', 'metadata'], allow_missing=True)
```
or something in that spirit

Reply:
rhs nicer to me (obv) but don't feel strongly, happy to revert